src/components/shared/SelectTopicsModal/SelectTopicsModal.js

```diff
@@ -0,0 +1,166 @@
+const SelectTopicsModal = ({ t, open, onClose, source, trigger }) => {
+  const [selected, setSelected] = useState([]);
+  const [match, setMatch]       = useState('');
+  const [name, setName]         = useState('');
+  const [alertMsg, setAlertMsg] = useState();
+
+  const areSourcesLoaded = useSelector(state => sourcesSelectors.areSourcesLoaded(state.sources));
+  const roots            = useSelector(state => selectors.getDisplayRoots(state.tags), isEqual) || [];
+  const getTagById       = useSelector(state => selectors.getTagById(state.tags));
+  const tree             = useMemo(() => getTree(roots, getTagById, null, t)[0], [roots.length, getTagById, t]);
+
+  const language = useSelector(state => settings.getLanguage(state.settings));
+  const dir      = getLanguageDirection(language);
+
+  const dispatch = useDispatch();
+
+  const create = () => {
+    const params = { name, tag_uids: selected, language, ...source };
+
+    dispatch(actions.add(MY_NAMESPACE_LABELS, params));
+  };
+
+  const clear = () => {
+    setSelected([]);
+    setName('');
+    setAlertMsg(null);
+  };
+
+  const renderColumn = col => (
+    <Grid.Column key={col.value}>
+      <Header
+        as="h2"
+        content={col.text}
+        className="topic_row_title topics__title"
+      />
+      {
+        col.children?.map(r => (
+          <TopicBranch
+            key={r.value}
+            match={match}
+            root={r}
+            selected={selected}
+            setSelected={setSelected}
+          />
+        ))
+      }
+    </Grid.Column>
+  );
+
+  const handleFilterChange = (e, data) => setMatch(data.value);
+
+  const handleSetName = (e, data) => setName(data.value);
+
+  const handleSave = () => {
+    create();
+    onClose();
+    setAlertMsg(t('personal.label.labelCreated'));
+  };
+
+  const handleCancel = () => {
+    onClose();
+    clear();
+  };
+
+  const needToLogin = NeedToLogin({ t });
+
+  return (
+    <>
+      <AlertModal message={alertMsg} open={!!alertMsg} onClose={clear} dir={dir} />
+      <Modal
+        open={open}
+        onClose={onClose}
+        className="select_topic_modal"
+        size="large"
+        trigger={trigger}
+        dir={dir}
+      >
+        <Modal.Header content={t('personal.label.header')} className="no-border" />
+        {
+          !!needToLogin ?
+            (
+              <Modal.Content>{needToLogin}</Modal.Content>
+            ) : (
+              <>
+                <Modal.Content style={{ paddingTop: 0 }}>
+                  <Input
+                    defaultValue={name}
+                    onChange={handleSetName}
+                    fluid
+                    className="label_name"
+                    error={selected.length > 0 && !name}
+                  >
+                    <Label
+                      content={t('personal.label.name')}
+                      basic
+                      className="no-border"
+                    />
+                    <input autoFocus />
```

Comment:
This should be prefilled in case we are generating a tag after a bookmark, as the bookmark title IS ACTUALLY the tagging/label name.

Reply:
Alex say that not